### PR TITLE
Allow `DefaultQubit2` to be initialized with a random seed

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Part of the effort to integrate the new device API. Support for sample-based measurements was recently added 🎉, and the API allowed for passing in a `np.Generator` object.

**Description of the Change:**
Add a `seed` keyword argument to the `__init__` method of `DefaultQubit2`. This is used to create a `np.Generator` object within the device upon initialization. This generator would then be passed to downstream calls of `simulate`.

**Benefits:**
Allows the user to seed the random number generator at the device level 🎉 

**Possible Drawbacks:**
None

### CODE REVIEW

**Comment by albi3ro** [comment]
[sc-38570]

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -56,6 +56,7 @@
 * Support for sample-based measurements has been added to the `DefaultQubit2` device.
   [(#4105)](https://github.com/PennyLaneAI/pennylane/pull/4105)
   [(#4114)](https://github.com/PennyLaneAI/pennylane/pull/4114)
+  [(#4120)](https://github.com/PennyLaneAI/pennylane/pull/4120)
```
Comment (rmoyard) [inline doc/releases/changelog-dev.md:59]: Minor: maybe I would describe the change for this specific PR as well

**Review by rmoyard — APPROVED** [review]
Thanks @eddddddy , it looks good to me :+1:

**Review by albi3ro — COMMENTED** [review]
I'd create one final test where we call `dev.execute` multiple times on the same device, and make sure we are getting different randomness each time.

We want to prevent something happening where the `rng` is reset upon each execution.

**Comment by timmysilv** [comment]
You make a compelling argument to keep it out of the device API, and probably equivalently out of the `ExecutionConfig`. Thanks for the details! This PR works for me 😄

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (eddddddy) [reply doc/releases/changelog-dev.md:64]: Added a separate entry 👍

**Review by albi3ro — APPROVED** [review]

**Pull request merged**